### ISSUE TITLE
replace path placeholder even if it's stored in a wide string

### ISSUE THREAD

**Issue by wojdyr** [conda/conda#4298]
Prefix placeholder can be stored in binary files as a wide string (wchar_t).
In such a case it's not replaced by conda.
I came across this problem when building wxWidgets that is using internally wide strings: UCS-2 on Windows and UCS-4 on Unix:
```
007c4f10  2f 00 00 00 6f 00 00 00  70 00 00 00 74 00 00 00  |/...o...p...t...|
007c4f20  2f 00 00 00 63 00 00 00  6f 00 00 00 6e 00 00 00  |/...c...o...n...|
007c4f30  64 00 00 00 61 00 00 00  2f 00 00 00 63 00 00 00  |d...a.../...c...|
007c4f40  6f 00 00 00 6e 00 00 00  64 00 00 00 61 00 00 00  |o...n...d...a...|
007c4f50  2d 00 00 00 62 00 00 00  6c 00 00 00 64 00 00 00  |-...b...l...d...|
007c4f60  2f 00 00 00 77 00 00 00  78 00 00 00 77 00 00 00  |/...w...x...w...|
007c4f70  69 00 00 00 64 00 00 00  67 00 00 00 65 00 00 00  |i...d...g...e...|
```
Perhaps conda could try to replace the placeholder in any of the popular encodings (UTF8, UTF16, UTF32)?

**Comment by kalefranz** [comment]
If you'd like to submit a pull request, the place to focus would be conda/core/portability.py.

**Comment by hmaarrfk** [comment]
Did you ever find a solution to this?

I'm currently building wxWidgets as well, and got into the same troubles.

**Comment by hmaarrfk** [comment]
Would a PR adding a loop to:
https://github.com/conda/conda/blob/master/conda/core/portability.py#L64

Be acceptable?

**Comment by hmaarrfk** [comment]
Unfortunately, it isn't as easy as that, one needs to ensure that the string doesn't accidentally become null terminated at the end.

See for example:

![image](https://user-images.githubusercontent.com/90008/82243947-44fc0d00-98f5-11ea-967c-2bea845f2905.png)

**Comment by hmaarrfk** [comment]
Yeah, the binary replacement path is a little trickier because you need to fixup the regex and string expansion.

**Comment by wojdyr** [comment]
@hmaarrfk  did you see PR #4307 linked above? I didn't follow up on that. But in short, replacing string is trivial, the question is how to minimize the overhead.

**Comment by jezdez** [comment]
@hmaarrfk Hey Mark, blast from the past, I'm not sure about this, do you happen to know if you're still affected by this?

**Comment by hmaarrfk** [comment]
has there been an explicit effort to search for wide strings? if not, then I presume I'm still affected. conda forge accepted my patch. maybe mamba will break installations though. I guess I should devote some time to actually fixing the problem and submitting a patch to mingwandroids specifications

**Comment by jezdez** [comment]
> has there been an explicit effort to search for wide strings? if not, then I presume I'm still affected. conda forge accepted my patch. maybe mamba will break installations though. I guess I should devote some time to actually fixing the problem and submitting a patch to mingwandroids specifications

Hey there, I was asking since I'm new to the project and have started to triage some of the existing but older ticket that didn't receive the attention in the past. To be clear, it's a legitimate reply that you don't know if this is fixed :) Just means that we can take a closer look again.

**Comment by hmaarrfk** [comment]
Not stale

**Comment by hmaarrfk** [comment]
Exciting!
